### DISCUSSION TITLE
End to end Template for running with web pack and HMR

### DISCUSSION THREAD

**Discussion by Isaac-Leonard** [RobinTail/express-zod-api#771] [Show and tell]
I've made a template repo for running this with a react frontend and Hot module reloading enabled for both the backend and frontend.
This means that you can modify routes and the server will update automatically so you can immediately test new routes.
HMR is also enabled for the frontend much like what create-react-appp does.
It also automatically updates the api client for the frontend whenever routes are changed so type mismatch errors will be instantly reported.
I hope someone finds this helpful.
https://github.com/Isaac-Leonard/E2E-HMR-Express-zod-api-template

**Comment by RobinTail** [comment]
Wow, thank you for publishing this, @Isaac-Leonard

**Comment by RobinTail** [comment]
Without `webpack` you can run `tsx` in watch mode:

```sh
yarn tsx --watch ./generate-client.ts
```
